src/app/shared/products-filter/products-filter.pipe.ts

```diff
@@ -0,0 +1,18 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'productsFilter',
+})
+export class ProductsFilterPipe implements PipeTransform {
+    transform<T extends {name: string}>(items: T[] | undefined | null, name: string): T[] | null | undefined {
+        if (!items?.length) {
+            return items;
+        }
+
+        const nameLowerCase = name.toLowerCase();
+
+        return items.filter(item => {
+            return (item['name'] as string).toLowerCase().includes(nameLowerCase);
```

Comment:
Рекомендация:
По идее можно было бы достать свойство черзе `.` т.к. мы уверены за счет типизации, что в `T` будет `name` - `<T extends {name: string}>`. 
```suggestion
            return item.name.toLowerCase().includes(nameLowerCase);
```